### ISSUE TITLE
Support testing for anonymous events

### ISSUE THREAD

**Issue by jdspugh** [NomicFoundation/hardhat#3219]
Contract:
```
contract C {
    event Update(uint256 n) anonymous;
    function cancel(uint256 n) external {
        ...
        emit Update(n);
    }
}
```
Text code:
```
await expect( c.connect(a).cancel(n) ).to.emit(c,'Update').withArgs(n)
```
The test code will work if Update is not anonymous. If it is anonymous this error will be output:
```
AssertionError: Expected event "Update" to be emitted, but it wasn't
```
Support for anonymous events would be great.

**Comment by fvictorio** [comment]
Makes sense, thanks for reporting this!

If I remember correctly, anonymous events cannot be filtered, so they can't really be distinguished from other anonymous events. This means that this would need a different API that just means "an anonymous event was emitted" and, in your example, `'Update'` wouldn't be used in the assertion. Something like:

```
await expect( c.connect(a).cancel(n) ).to.emitAnonymous().withArgs(n)
```

Do you think that would be acceptable?

**Comment by jdspugh** [comment]
Yes, that makes sense also. Thank you for considering this request.
